src/main/java/kitchenpos/domain/Menu.java

```diff
@@ -1,63 +1,99 @@
 package kitchenpos.domain;
 
+import javax.persistence.*;
 import java.math.BigDecimal;
 import java.util.List;
+import java.util.Objects;
 
+import static kitchenpos.constants.ErrorCodeType.MENU_PRICE_NOT_OVER_SUM_PRICE;
+
+@Entity
 public class Menu {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
-    private BigDecimal price;
-    private Long menuGroupId;
-    private List<MenuProduct> menuProducts;
+    @Embedded
+    private Price price;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    private MenuGroup menuGroup;
 
-    public Menu() {
+    @Embedded
+    private MenuProducts menuProducts = new MenuProducts();
+
+    protected Menu() {
     }
 
-    public Menu(Long id, String name, BigDecimal price, Long menuGroupId, List<MenuProduct> menuProducts) {
+    public Menu(Long id, String name, Price price, MenuGroup menuGroup) {
         this.id = id;
         this.name = name;
         this.price = price;
-        this.menuGroupId = menuGroupId;
-        this.menuProducts = menuProducts;
+        this.menuGroup = menuGroup;
     }
 
-    public Long getId() {
-        return id;
+    public Menu(String name, Price price, MenuGroup menuGroup) {
+        this.name = name;
+        this.price = price;
+        this.menuGroup = menuGroup;
     }
 
-    public void setId(final Long id) {
-        this.id = id;
+
+    private static void validCheckMeuProductPrice(List<MenuProduct> menuProducts, Price price) {
+        BigDecimal sum = BigDecimal.ZERO;
+        for (MenuProduct menuProduct : menuProducts) {
+            sum = sum.add(menuProduct.getProduct()
+                    .getPrice().multiply(BigDecimal.valueOf(menuProduct.getQuantity())));
+        }
+
+        validCheckMenuPrice(sum, price);
     }
 
-    public String getName() {
-        return name;
+    private static void validCheckMenuPrice(BigDecimal sumPrice, Price requestPrice) {
+        if (requestPrice.getPrice().compareTo(sumPrice) > 0) {
+            throw new IllegalArgumentException(MENU_PRICE_NOT_OVER_SUM_PRICE.getMessage());
+        }
     }
 
-    public void setName(final String name) {
-        this.name = name;
+    public void addMenuProducts(MenuProducts menuProducts) {
+        validCheckMeuProductPrice(menuProducts.getList(), price);
```

Comment:
```suggestion
        validCheckMeuProductPrice(menuProducts, price);
```
getter를 쓰지않고 `menuProducts`가 총 금액 합을 응답할수 있지 않을까요? 🤔
`Price sum = menuProducts.price();`